packages/backend/src/modules/tracked-txs/modules/l2-costs/api/L2CostsController.test.ts

```diff
@@ -0,0 +1,362 @@
+import {
+  TrackedTxsConfigRecord,
+  TrackedTxsConfigsRepository,
+} from '../../../repositories/TrackedTxsConfigsRepository'
+import { TrackedTxId } from '../../../types/TrackedTxId'
+import {
+  L2CostsRecord,
+  L2CostsRepository,
+} from '../repositories/L2CostsRepository'
+import { DetailedTransaction, L2CostsController } from './L2CostsController'
+
+const NOW = UnixTime.now()
+
+describe(L2CostsController.name, () => {
+  describe(L2CostsController.prototype.getL2Costs.name, () => {
+    it('correctly calculates l2costs', async () => {
+      const l2CostsRepository = mockObject<L2CostsRepository>({
+        getByProjectSinceTimestamp: mockFn().resolvesTo([]),
+      })
+      const controller = getMockL2CostsController({
+        projects: MOCK_PROJECTS,
+        trackedTxsConfigsRepository: mockObject<TrackedTxsConfigsRepository>({
+          getByProjectIdAndType: mockFn()
+            .given(MOCK_PROJECTS[1].projectId, 'l2costs')
+            .resolvesToOnce([
+              mockObject<TrackedTxsConfigRecord>({
+                id: TrackedTxId.unsafe('aaa'),
+                projectId: MOCK_PROJECTS[1].projectId,
+                untilTimestampExclusive: undefined,
+                lastSyncedTimestamp: new UnixTime(1000),
+              }),
+            ])
+            .given(MOCK_PROJECTS[2].projectId, 'l2costs')
+            .resolvesToOnce([
+              mockObject<TrackedTxsConfigRecord>({
+                id: TrackedTxId.unsafe('bbb'),
+                projectId: MOCK_PROJECTS[2].projectId,
+                untilTimestampExclusive: undefined,
+                lastSyncedTimestamp: new UnixTime(2000),
+              }),
+            ]),
+        }),
+        l2CostsRepository,
+      })
+
+      controller.makeTransactionCalculations = mockFn().returns([
+        mockObject<DetailedTransaction>({
+          timestamp: NOW.add(-1, 'hours'),
+        }),
+        mockObject<DetailedTransaction>({ timestamp: NOW.add(-2, 'hours') }),
+      ])
+
+      controller.sumDetails = mockFn().returns(mockObject<L2CostsDetails>({}))
+
+      const result = await controller.getL2Costs()
+
+      // filters out projects without trackedTxsConfig
+      expect(
+        l2CostsRepository.getByProjectSinceTimestamp,
+      ).toHaveBeenCalledTimes(2)
+      expect(result.type).toEqual('success')
+      expect(result.data.projects).toEqual({
+        project2: {
+          syncedUntil: new UnixTime(1000),
+          last24h: mockObject<L2CostsDetails>({}),
+          last7d: mockObject<L2CostsDetails>({}),
+          last30d: mockObject<L2CostsDetails>({}),
+          last90d: mockObject<L2CostsDetails>({}),
+        },
+        project3: {
+          syncedUntil: new UnixTime(2000),
+          last24h: mockObject<L2CostsDetails>({}),
+          last7d: mockObject<L2CostsDetails>({}),
+          last30d: mockObject<L2CostsDetails>({}),
+          last90d: mockObject<L2CostsDetails>({}),
+        },
+      })
+    })
```

Comment:
lack of:
* `expect(makeTransactionCalculations).toHaveBeenCalledWith(...)`
* `expect(sumDetails).toHaveBeenCalledWith(...)`